azkaban-web-server/src/main/java/azkaban/flowtrigger/DependencyInstanceContext.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.flowtrigger;
+
+/**
+ * This interface provides getExecId() method
+ * to get execution id for a running dependency instance.
+ * The implementation classes should hold context
+ * information for a running dependency instance.
+ */
+public interface DependencyInstanceContext {
+
+  int getExecId();
```

Comment:
How is this id used?

Reply:
it's used to identify a dependency instance. Just like flow execution id.